foodbornenyc/sources/twitter_search.py

```diff
@@ -154,8 +161,10 @@ def query_twitter(how_long=0, interval=5):
             continue
         #logger.info("%i Total unique tweets in %i:%i:%i time", len(id_set),
         #            *sec_to_hms(time()-start))
-        new_tweets = [ t for t in tweets if t['id_str'] not in id_set ]
-        new_Tweets = tweets_to_Tweets(new_tweets, fields)
+        new_tweets = [t for t in tweets if t['id_str'] not in id_set]
+        # if a retrieved tweet has a loc/user with a matching ID already in the
+        # db, that loc/user is updated instead of a new one added, bc of merge
+        new_Tweets = [db.merge(t) for t in tweets_to_Tweets(new_tweets, fields)]
```

Comment:
Hmm... I'm surprised my test passed because both canned tweets have the same user. [It looks like](http://docs.sqlalchemy.org/en/rel_1_0/orm/session_state_management.html#merging) `db.merge` automatically merges uncommitted things in the same session. If the flag `load=True`, it also looks in the db.